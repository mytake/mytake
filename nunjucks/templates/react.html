{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
