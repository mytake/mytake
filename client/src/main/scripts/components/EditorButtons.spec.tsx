/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2018 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import EditorButtons from "./EditorButtons";

jest.mock("./Banner");

const eventHandlers = {
  handleDeleteClick: jest.fn(),
  handlePublishClick: jest.fn(),
  handleSaveClick: jest.fn(),
};

test("Initial status", () => {
  const status = {
    saved: true,
    saving: false,
    error: false,
    message: "",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Unsaved", () => {
  const status = {
    saved: false,
    saving: false,
    error: false,
    message: "",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Saving", () => {
  const status = {
    saved: false,
    saving: true,
    error: false,
    message: "",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Publishing", () => {
  const status = {
    saved: false,
    saving: true,
    error: false,
    message: "Publishing Take.",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Publishing error", () => {
  const status = {
    saved: false,
    saving: false,
    error: true,
    message: "There was an error publishing your Take.",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Title too long", () => {
  const status = {
    saved: false,
    saving: false,
    error: true,
    message: "Title cannot be longer than 255 characters.",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Saving", () => {
  const status = {
    saved: false,
    saving: true,
    error: false,
    message: "Saving Take.",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Save successful", () => {
  const status = {
    saved: true,
    saving: false,
    error: false,
    message: "Save successful!",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Unknown error", () => {
  const status = {
    saved: false,
    saving: true,
    error: true,
    message: "There was an error modifying your Take.",
  };
  const tree = renderer
    .create(<EditorButtons status={status} eventHandlers={eventHandlers} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
