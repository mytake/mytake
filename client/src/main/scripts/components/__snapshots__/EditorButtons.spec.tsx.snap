// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial status 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    
  </Banner>
  <button
    className="editor__button editor__button--disabled"
    disabled={true}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Publishing 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    Publishing Take.
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Publishing error 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={false}
  >
    There was an error publishing your Take.
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Save successful 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    Save successful!
  </Banner>
  <button
    className="editor__button editor__button--disabled"
    disabled={true}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Saving 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Saving 2`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    Saving Take.
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Title too long 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={false}
  >
    Title cannot be longer than 255 characters.
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Unknown error 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={false}
  >
    There was an error modifying your Take.
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;

exports[`Unsaved 1`] = `
<div
  className="editor__meta"
>
  <Banner
    isSuccess={true}
  >
    
  </Banner>
  <button
    className="editor__button editor__button--save"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
  <button
    className="editor__button editor__button--publish"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save & Publish
  </button>
  <button
    className="editor__button editor__button--delete"
    disabled={false}
    onClick={[MockFunction]}
  >
    Delete Draft
  </button>
</div>
`;
